### PR TITLE
[Backend] Allow making submissions private before finished

### PR DESCRIPTION
**Pull request by Ram81**
### Description

This PR adds changes to allow making submissions private before it is in `FINISHED` state. Earlier we only allowed modifying submission visibilty when submission was evaluated successfully

### CODE REVIEW

File: frontend/src/js/controllers/challengeCtrl.js [hunk 1]
```diff
@@ -530,6 +531,11 @@
                     formData.append("project_url", vm.projectUrl);
                     formData.append("publication_url", vm.publicationUrl);
                     formData.append("submission_metadata", JSON.stringify(vm.metaAttributesforCurrentSubmission));
+                    console.log("is public : " + vm.isPublicSubmission);
```
Comment (RishabhJain2018) [inline frontend/src/js/controllers/challengeCtrl.js]: Can we please remove the console statements?

File: frontend/src/js/controllers/challengeCtrl.js [hunk 2]
```diff
@@ -530,6 +531,11 @@
...
+                    console.log("is public =null : " + vm.isPublicSubmission === null)
```
Comment (RishabhJain2018) [inline frontend/src/js/controllers/challengeCtrl.js]: Same as above

**Review by RishabhJain2018 — CHANGES_REQUESTED** [review]
Can you please post a UI screenshot for the submit Tab.

**Comment by Ram81** [comment]
Screen Shot 2020-10-06 at 11 21 06 PM

**Review by RishabhJain2018 — APPROVED** [review]

**Pull request merged**